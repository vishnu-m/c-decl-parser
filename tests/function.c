/* Declaration with static storage class specifier */
extern int max(int a, int b);

/* Return type is pointer to array of 3 int */
int (*foo(const void *p))[3]; 

/* Declares function of type double(void) */
static double const check(void) { return 0.; } 

inline int sum(int a, int b) 
{
        return a + b;
}

/* Declaration with definition */
int increment(int a){           
        a++;
        return a;
}

