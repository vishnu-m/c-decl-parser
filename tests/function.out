FunctionDecl	"max"		storage class specifier = "extern"		return type = "int"	Args = 
		"a"		type="int"
		"b"		type="int"
FunctionDecl	"foo"		storage class specifier = ""		return type = "int (*)[3]"	Args = 
		"p"		type="const void *"
FunctionDecl	"check"		storage class specifier = "static"		return type = "const double"	Args = 
FunctionDecl			 inline 		"sum"		storage class specifier = ""		return type = "int"	Args = 
		"a"		type="int"
		"b"		type="int"
FunctionDecl	"increment"		storage class specifier = ""		return type = "int"	Args = 
		"a"		type="int"
